### PR TITLE
RELAY-460 test: added new tests internal/usecase/api-server

### PR DESCRIPTION
**Pull request by ALSymbiotic**
### **PR Type**
Tests

___

### **Description**
- Added comprehensive unit tests for 8 gRPC API handler methods

- Tests cover success cases, error handling, and edge cases

- Added error handling interceptor tests for gRPC error conversion

- Added max streams limit tests for streaming endpoints

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["API Handler Methods"] -->|tested by| B["Unit Tests"]
  B --> C["Success Cases"]
  B --> D["Error Handling"]
  B --> E["Edge Cases"]
  F["Error Interceptors"] -->|tested by| G["Interceptor Tests"]
  H["Streaming Endpoints"] -->|tested by| I["Max Streams Tests"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td><strong>get_aggregation_proof_v1_test.go</strong><dd><code>Tests for GetAggregationProof handler method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-52250f1384783cb4e3491218454546191b3a9ce4404ada623365d403e24dbaab">+121/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_aggregation_status_v1_test.go</strong><dd><code>Tests for GetAggregationStatus handler with sorting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-472cdd8ec1fe197529f12902fe00696c8702b9f4439df66ed26b93a1f4b4af5e">+147/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_current_epoch_v1_test.go</strong><dd><code>Tests for GetCurrentEpoch handler method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-202275e09cd3bd752be41bdeddf9b6194ac23eb87a3d9829f55363c95c7cf945">+114/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_last_all_committed_v1_test.go</strong><dd><code>Tests for GetLastAllCommitted handler method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-ba235273fd00e00388b28e8435bba8178bd2af7942da24e31eee3eef25ee67fa">+216/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_last_committed_v1_test.go</strong><dd><code>Tests for GetLastCommitted handler with validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-2aa53f3e60a113e794dcf9ded263b43508476b8da86ea6d8e783616e30efd77e">+287/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_signatures_v1_test.go</strong><dd><code>Tests for GetSignatures handler method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-707046605e758b55d9941c2d0a2aa9fd109a2e20c8099bf81828681ef79d1b45">+132/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_validator_set_v1_test.go</strong><dd><code>Tests for GetValidatorSet and status conversion</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-8faef8a4c5ba26f2f89f2b9beac107dc63b429bf2a41406cbdb17c656a0ceb4c">+190/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_validatorset_metadata_v1_test.go</strong><dd><code>Tests for GetValidatorSetMetadata handler method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-0bb37331bd0d4b8e8f338ad7ca46601da39055c4b7c78c20f900bc57ef0260d6">+196/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>interceptors_test.go</strong><dd><code>Tests for error handling and gRPC interceptors</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-00c145cd8d68ab1299b76ecedc3a894c2bdb1d807d6d64feb37a9bfdb34fbf2f">+247/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>listen_proofs_v1_test.go</strong><dd><code>Added max streams limit test for proofs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-e3d4717f25073c6e87ec941c63a5be1d04bb69efc112b3d13691a0e6294c4dca">+20/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>listen_signatures_v1_test.go</strong><dd><code>Added max streams limit test for signatures</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-9530dd338fa11e7ffe12085cb6a0d6d939ba8742928b1ed4d20f12241ee7ec8d">+20/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>listen_validator_set_v1_test.go</strong><dd><code>Added max streams limit test for validator sets</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-02b53852d9254c1959e2559899efaf58b5f9ed8accdcf0fcf2fbff279b23c906">+20/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>sign_message_v1_test.go</strong><dd><code>Tests for SignMessage handler method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/264/files#diff-298cea4370e5cc639ed97336209a25d8b052070cc2974c9c0f02f43f467ec8f3">+180/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: internal/usecase/api-server/get_aggregation_proof_v1_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	defer ctrl.Finish()
+
+	mockRepo := mocks.NewMockrepo(ctrl)
+	handler := &grpcHandler{
+		cfg: Config{
+			Repo: mockRepo,
+		},
+	}
+
+	ctx := context.Background()
```
Comment (ilyasymbiotic) [inline internal/usecase/api-server/get_aggregation_proof_v1_test.go]: I think in other codebase we decided to use t.Context()
